kubelet/datadog_checks/kubelet/summary.py

```diff
@@ -0,0 +1,149 @@
+            if container_name is None:
+                self.log.warning("Kubelet reported stats without container name for pod: %s/%s", pod_namespace, pod_name)
+                continue
+
+            # No mistake, we need to give a tuple as parameter
+            container_id = pod_list_utils.get_cid_by_name_tuple((pod_namespace, pod_name, container_name))
+            if container_id is None:
+                self.log.debug("Container id not found from /pods for container: %s/%s/%s - no metrics will be sent", pod_namespace, pod_name, container_name)
+                continue
+
+            # TODO: In `containers` we also have terminated init-containers, probably to be excluded?
+            if pod_list_utils.is_excluded(container_id):
+                continue
+
+            # Finally, we can get tags for this container
+            container_tags = tags_for_container(replace_container_rt_prefix(container_id), tagger.HIGH)
+            if not container_tags:
+                self.log.debug("Tags not found for container: %s/%s/%s:%s - no metrics will be sent", pod_namespace, pod_name, container_name, container_id)
+            container_tags += instance_tags
+
+            cpu = container.get('cpu')
+            if cpu:
+                cpu_total = cpu.get('usageCoreNanoSeconds')
+                if cpu_total:
+                    self.rate(self.NAMESPACE + '.cpu.usage.total', cpu_total, container_tags)
+
+            memory = container.get('memory')
+            if memory:
+                working_set = memory.get('workingSetBytes')
+                if working_set:
+                    self.gauge(self.NAMESPACE + '.memory.working_set', working_set, container_tags)
+
+            rootfs = container.get('rootfs')
+            if rootfs:
+                # TODO: Review meaning of these metrics as capacity != available + used
+                # availableBytes = rootfs.get('availableBytes')
+                capacity_bytes = rootfs.get('capacityBytes')
+                used_bytes = rootfs.get('usedBytes')
+                if used_bytes:
+                    self.gauge(self.NAMESPACE + '.filesystem.usage', used_bytes, container_tags)
+                if used_bytes and capacity_bytes:
+                    self.gauge(self.NAMESPACE + '.filesystem.usage_pct', float(used_bytes) / capacity_bytes, container_tags)
+
+    def _report_system_container_metrics(self, stats, instance_tags):
+        sys_containers = stats.get('node', {}).get('systemContainers', [])
+        for ctr in sys_containers:
+            if ctr.get('name') == 'runtime':
+                mem_rss = ctr.get('memory', {}).get('rssBytes')
+                if mem_rss:
+                    self.gauge(self.NAMESPACE + '.runtime.memory.rss', mem_rss, instance_tags)
+                cpu_usage = ctr.get('cpu', {}).get('usageNanoCores')
+                if cpu_usage:
+                    self.gauge(self.NAMESPACE + '.runtime.cpu.usage', cpu_usage, instance_tags)
```

Comment:
why do we submit `cpu.usage` metrics as gauges here as opposed to what's done in `_report_container_stats` submitted as rates? is this on purpose?

Reply:
It's using `usageNanoCores` which is the "instant" usage computed by Kubelet itself while I'm using `usageCoreNanoSeconds` which is a rate.
Maybe we should align both.
Note that code was already existing (I just moved it) but `kubernetes.cpu.usage.total` is a rate, so I believe the `.kubelet` one should have been a rate too.

Reply:
okay so this is not in the scope of this PR, let's investigate/fix that later